apps/aecontract/src/aect_sophia.erl

```diff
@@ -22,34 +21,31 @@
 -type wrapped_code() :: #{ source_hash := aec_hash:hash()
                          , type_info   := [{binary(), binary(), binary(), binary()}]
                          , byte_code   := binary()
-                         , compiler_version := binary()
+                         , compiler_version => binary()
                          , contract_vsn := integer()
+                         , abi_version => integer()
                          }.
 
 -export_type([ wrapped_code/0 ]).
 
-%% After hard fork Minerva, we accept ?SOPHIA_CONTRACT_VSN_2 in serialization
--define(SOPHIA_CONTRACT_VSN, ?SOPHIA_CONTRACT_VSN_2).
-
 is_legal_serialization_at_height(?SOPHIA_CONTRACT_VSN_1, _Height) ->
     true;
 is_legal_serialization_at_height(?SOPHIA_CONTRACT_VSN_2, Height) ->
-    aec_hard_forks:protocol_effective_at_height(Height) >= ?MINERVA_PROTOCOL_VSN.
-
-serialize(Map) ->
-    serialize(Map, ?SOPHIA_CONTRACT_VSN).
+    aec_hard_forks:protocol_effective_at_height(Height) >= ?MINERVA_PROTOCOL_VSN;
+is_legal_serialization_at_height(?SOPHIA_CONTRACT_VSN_3, Height) ->
```

Comment:
Ideally this shall move away from assuming protocol known at height. Though orthogonal.